frontend/src/metabase/visualizations/components/LineAreaBarChart.jsx

```diff
@@ -259,11 +261,45 @@ export default class LineAreaBarChart extends Component {
     return settings;
   }
 
+  getLegendSettings() {
+    const {
+      card,
+      series,
+      settings,
+      showTitle,
+      onChangeCardAndRun,
+    } = this.props;
+
+    const title = settings["card.title"] || card.name;
+    const description = series["card.description"];
+    const data = series._raw || series;
+    const cardIds = new Set(data.map(s => s.card.id));
```

Comment:
If all we will use `cardIds` for is checking its size, do we need to make it a Set?

Wouldn't it be faster, all things considered, if we just did

```js
const cardIds = data.map(s => s.card.id);

// and below

const canSelectTitle = cardIds.length === 1 && onChangeCardAndRun;

```

?

Reply:
…we could also make a little function that loops and immediately breaks out of the loop returning `false` if the count passes 1.

Reply:
We cannot do this:
```
const cardIds = data.map(s => s.card.id);
const canSelectTitle = cardIds.length === 1 && ...;
```

because there might be duplicate card ids, and we need to check if there are unique cards (questions).